### PR TITLE
Fix regression in assigned geometries for iCub 2.* models

### PR DESCRIPTION
**Pull request by traversaro**
Regression introduced in https://github.com/robotology/icub-models-generator/pull/165 . Fix https://github.com/robotology/icub-models/issues/68 .

### CODE REVIEW

**Comment by traversaro** [comment]
To clarify, this if/then logic in CMake is clearly madness, so any proposal to have more mantainable yaml files is welcome.

File: simmechanics/CMakeLists.txt [hunk 1]
```diff
@@ -158,8 +160,6 @@ macro(generate_icub_simmechanics)
       size: 0.117 0.100 0.006
       origin: \"0.0 0.0 0.003 0.0 0.0 0.0\"
 ")
-    else()
-      set(ASSIGNED_COLLISION_GEOMETRIES "")
     endif()
```
Comment (prashanthr05) [inline simmechanics/CMakeLists.txt:163]: Ahh, oops! I did something bad there.

Thanks for the fix!! @traversaro

**Comment by prashanthr05** [comment]
> To clarify, this if/then logic in CMake is clearly madness, so any proposal to have more mantainable yaml files is welcome.

I wouldn't say it is a total madness. As a CMake beginner, I failed to pay attention to the simpler logic while focusing on the actual feature addition.  My bad!

**Review by prashanthr05 — APPROVED** [review]

**Review by GiulioRomualdi — APPROVED** [review]
Thank you @traversaro

**Comment by traversaro** [comment]
> > To clarify, this if/then logic in CMake is clearly madness, so any proposal to have more mantainable yaml files is welcome.
> 
> I wouldn't say it is a total madness. As a CMake beginner, I failed to pay attention to the simpler logic while focusing on the actual feature addition. My bad!


To be honest that was kind of a general comment, not related to this regression.

**Pull request merged**